shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -470,147 +784,448 @@ object Types {
     */
   case class AppliedType(tycon: Type, args: List[Type]) extends TypeProxy with ValueType {
     override def underlying(using Context): Type = tycon
+
+    override def superType(using Context): Type =
+      tycon match
+        //case tycon: HKTypeLambda => defn.AnyType
+        case tycon: TypeRef if tycon.symbol.isClass => tycon
+        case tycon: TypeProxy                       => tycon.superType.applyIfParameterized(args)
+        case _                                      => AnyType
+
+    def tyconTypeParams(using Context): List[ParamInfo] =
+      val tparams = tycon.typeParams
+      /*if (tparams.isEmpty) HKTypeLambda.any(args.length).typeParams else*/
+      tparams
+
+    /** Is this an unreducible application to wildcard arguments?
+      * This is the case if tycon is higher-kinded. This means
+      * it is a subtype of a hk-lambda, but not a match alias.
+      * (normal parameterized aliases are removed in `appliedTo`).
+      * Applications of higher-kinded type constructors to wildcard arguments
+      * are equivalent to existential types, which are not supported.
+      */
+    def isUnreducibleWild(using Context): Boolean =
+      // TODO tycon.isLambdaSub && hasWildcardArg && !isMatchAlias
+      false
+
+    override def findMember(name: Name, pre: Type)(using Context): Symbol =
+      tycon match
+        case tycon: TypeRef =>
+          if tycon.resolveToSymbol.isClass then tycon.findMember(name, pre)
+          else ???
+        case _ =>
+          ???
+
+    def derivedAppliedType(tycon: Type, args: List[Type])(using Context): AppliedType =
+      if ((tycon eq this.tycon) && (args eq this.args)) this
+      else AppliedType(tycon, args)
+
+    def map(op: Type => Type)(using Context): AppliedType =
+      derivedAppliedType(op(tycon), mapConserve(args)(op))
   }
 
+  private def mapConserve[T <: AnyRef](xs: List[T])(op: T => T): List[T] =
```

Comment:
blows stack

Reply:
Actually, List now has a mapConserve method in the std library - should probably use this